### DISCUSSION TITLE
Facet frame only for "whole" plot

### DISCUSSION THREAD

**Discussion by Hvass-Labs** [observablehq/plot#1951] [Q&A]
This issue is moved from deep within the comments of https://github.com/observablehq/plot/discussions/1943.

When using a frame and facets together, the frame is added to every single facet (or sub-plot). How can I add the frame to only the "whole" plot?

The plotting code is essentially:

    Plot.plot({
      grid: true,
      fx: {interval: "3 months"},
      x: {axis: null, padding: 0},
      y: {label: null},
      color: {legend: true},
      marks: [
        Plot.barY(data, {fx: "Date", y: "Value", x: "Name", fill: "Name"}),
      ]
    });

I have tried two ways of adding the frame:

    Plot.frame().plot({ ... });

and:
    
    Plot.plot({marks:[Plot.frame(), ...]});    

Both ways produce the following plot with a frame on every facet.

I think there is a similar problem when using the text mark to add a title to the plot, where it is also added to every facet. (This is a hack to circumvent how you combine HTML and SVG code which causes problems for my application.)

Thanks!

![Faceted bar-plots](https://github.com/observablehq/plot/assets/13588114/cdb2ecfd-05e5-43e2-b1fd-7862b587a980)

**Comment by Fil** [comment]
For the frame, that's what the mark is and I don't think there is a way around it—we would have to create a specific mark or option for that. See https://observablehq.com/@recifs/plot-facet-encompassing-frame--dev

For the text, you could filter it by adding an option, either:
* `facetAnchor: "left"` (or more generically `facetAnchor: "top-left"`)
* or `fx: [x]` where x corresponds to the facet you want to see the text.

**Reply by Hvass-Labs** [reply]
Thanks very much for the detailed code-sample, that was extremely quick!

I think it would be very helpful to the user if you make a built-in mark for this, so users don't have to copy-paste all that code.

Please keep this issue open so you know there is a request for this.
